### PR TITLE
Cleanup a few forced downcasts

### PR DESCRIPTION
**Pull request by nheagy**
I fixed a few of our uses of `as!` to prove out my thoughts while writing [a style guide proposal](https://github.com/wordpress-mobile/swift-style-guide/issues/7)

**To test:**

Functionality should not have changed. If account settings and comment sorting still work, then that would appear to be the case.

_For authoring the first accepted style guide proposal:_

Needs review: @diegoreymendez

### CODE REVIEW

File: WordPress/Classes/Services/AccountSettingsService.swift [hunk 1]
```diff
@@ -159,18 +159,18 @@ class AccountSettingsService {
         let request = NSFetchRequest(entityName: ManagedAccountSettings.entityName)
         request.predicate = NSPredicate(format: "account.userID = %d", userID)
         request.fetchLimit = 1
-        let results = (try? context.executeFetchRequest(request) as! [ManagedAccountSettings]) ?? []
+        let results = (try? context.executeFetchRequest(request)) as? [ManagedAccountSettings] ?? []
```
Comment (diegoreymendez) [inline WordPress/Classes/Services/AccountSettingsService.swift]: Nice job on this change.  However I feel this code is suffering in readability due to too much stuff going on in a single line.

Can we go for something like this instead? (this is untested, but just to convey the idea)

```
guard let results = (try? context.executeFetchRequest(request)) as? [ManagedAccountSettings] else {
    return nil
}

return results.first
```

File: WordPress/Classes/Services/AccountSettingsService.swift [hunk 2]
```diff
@@ -159,18 +159,18 @@ class AccountSettingsService {
...
         return results.first
     }
 
     private func createAccountSettings(userID: Int, settings: AccountSettings) {
         let accountService = AccountService(managedObjectContext: context)
-        guard let account = accountService.findAccountWithUserID(userID) else {
+        guard let account = accountService.findAccountWithUserID(userID),
+        let managedSettings = NSEntityDescription.insertNewObjectForEntityForName(ManagedAccountSettings.entityName, inManagedObjectContext: context) as? ManagedAccountSettings else {
```
Comment (diegoreymendez) [inline WordPress/Classes/Services/AccountSettingsService.swift]: The error message shown inside the guard only really makes sense if the first part of the guard fails.

I'd recommend a separate guard in this case for the second condition.

**Comment by nheagy** [comment]
Alright @diegoreymendez I agree with both your changes. Here's a new push, what do you think?

**Comment by diegoreymendez** [comment]
:shipit:

**Comment by nheagy** [comment]
Thanks @diegoreymendez !

**Pull request merged**